src/explore-education-statistics-admin/src/pages/release/datablocks/components/ChartDataSetsConfiguration.tsx

```diff
@@ -0,0 +1,253 @@
+import Button from '@common/components/Button';
+import ButtonText from '@common/components/ButtonText';
+import { Form, FormFieldSelect, FormSelect } from '@common/components/form';
+import VisuallyHidden from '@common/components/VisuallyHidden';
+import { DataSet } from '@common/modules/charts/types/dataSet';
+import expandDataSet from '@common/modules/charts/util/expandDataSet';
+import generateDataSetKey from '@common/modules/charts/util/generateDataSetKey';
+import { LocationFilter } from '@common/modules/table-tool/types/filters';
+import { FullTableMeta } from '@common/modules/table-tool/types/fullTable';
+import { Dictionary } from '@common/types';
+import Yup from '@common/validation/yup';
+import { Formik } from 'formik';
+import difference from 'lodash/difference';
+import mapValues from 'lodash/mapValues';
+import React, { ReactNode, useMemo } from 'react';
+
+const formId = 'chartDataSetsConfigurationForm';
+
+interface FormValues {
+  filters: Dictionary<string>;
+  indicator: string;
+  location: string;
+  timePeriod: string;
+}
+
+interface Props {
+  buttons?: ReactNode;
+  dataSets?: DataSet[];
+  forms: ChartBuilderForms;
+  isSaving?: boolean;
+  meta: FullTableMeta;
+  onChange: (dataSets: DataSet[]) => void;
+  onSubmit: () => void;
+}
+
+const ChartDataSetsConfiguration = ({
+  buttons,
+  isSaving,
+  forms,
+  meta,
+  dataSets = [],
+  onChange,
+  onSubmit,
+}: Props) => {
+  const indicatorOptions = useMemo(() => Object.values(meta.indicators), [
+    meta.indicators,
+  ]);
+
+  const locationOptions = useMemo(
+    () =>
+      meta.locations.map(location => ({
+        value: location.id,
+        label: location.label,
+      })),
+    [meta.locations],
+  );
+
+  return (
+    <>
+      <Formik<FormValues>
+        initialValues={{
+          filters: mapValues(meta.filters, filterGroup =>
+            filterGroup.options.length === 1
+              ? filterGroup.options[0].value
+              : '',
+          ),
+          indicator:
+            meta.indicators.length === 1 ? meta.indicators[0].value : '',
+          location: meta.locations.length === 1 ? meta.locations[0].id : '',
+          timePeriod:
+            meta.timePeriodRange.length === 1
+              ? meta.timePeriodRange[0].value
+              : '',
+        }}
+        validateOnBlur={false}
+        validateOnChange={false}
+        validationSchema={Yup.object<FormValues>({
+          indicator: Yup.string().required('Select an indicator'),
+          filters: Yup.object(
+            mapValues(meta.filters, (filter, category) =>
+              Yup.string().required(`Select a ${category.toLowerCase()}`),
+            ),
+          ),
+          location: Yup.string(),
+          timePeriod: Yup.string(),
+        })}
+        onSubmit={values => {
+          const { indicator } = values;
+          const filters = Object.values(values.filters);
+
+          const timePeriod: DataSet['timePeriod'] = values.timePeriod
```

Comment:
We've done this to avoid setting empty strings in the data set. The initial form values for location/time period are actually empty strings, even though they're technically optional. This is because Formik's types force you to set initial values for all fields (even optional ones).

I've left a comment here for clarity anyway :smile: